network/agent/agent.ts

```diff
@@ -23,16 +24,27 @@ export function getAgent (uri: string, opts: AgentOptions) {
 
 function getNonProxyAgent (uri: string, opts: AgentOptions) {
   const parsedUri = new URL(uri)
+  const host = nerfDart(uri)
   const isHttps = parsedUri.protocol === 'https:'
 
+  const clientCertificates = opts.clientCertificates?.[host] ?? null
```

Comment:
maybe just do something like
```suggestion
  const { ca, cert, key } = {
    ...opts,
    ...opts.clientCertificates?.[host],
  }
```